flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/partitioner/CustomPartitionerWrapper.java

```diff
@@ -55,6 +56,14 @@ public int selectChannel(SerializationDelegate<StreamRecord<T>> record) {
 		return partitioner.partition(key, numberOfChannels);
 	}
 
+	@Override
+	public ChannelStateRescaler getDownstreamChannelStateRescaler() {
+		// fully rely on filtering downstream
+		// note that custom partitioners are not officially supported - the user has to force rescaling
+		// in that case, we assume that the custom partitioner is deterministic
+		return ChannelStateRescaler.BROADCAST;
```

Comment:
This will not distribute state to the newer tasks in case of up-scaling, right?
I think it can cause problems if the user-supplied partitioner relies on the new DOP.

OTH, if not supported officially, we can leave it as it.

WDYT?

Reply:
`BROADCAST` would replicate each channel state to each subtask (very costly) including all new states. All superfluous replicas are filtered out downstream by the partitioner.
Note that I will extend your check for cyclic graphs to customer partitioners for disabling UC. So the user has to force it to make it work. However, I do not see a reason to then fail on the second level again.

Reply:
Yes, you are right.